pkg/config/config.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2015 flannel authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/flannel-io/flannel/pkg/config/etcd"
+	"github.com/flannel-io/flannel/pkg/config/kube"
+)
+
+type SubnetManagerType uint
+
+const (
+	EtcdType SubnetManagerType = iota
+	KubeType
+)
+
+type Config struct {
+	etcd.EtcdSubnetMngrConfig
+	kube.KubeSubnetMngrConfig
```

Comment:
Instead of this, I would add all the "official" config parameters that we document in the [docs](https://github.com/flannel-io/flannel/blob/master/Documentation/configuration.md#configuration) here:

```
type Config struct {
	EnableIPv4  bool
	EnableIPv6  bool
	BackendType string          `json:"-"`
	Backend     json.RawMessage `json:",omitempty"`
	Network       ip.IP4Net
	IPv6Network   ip.IP6Net
	SubnetMin     ip.IP4
	SubnetMax     ip.IP4
	IPv6SubnetMin *ip.IP6
	IPv6SubnetMax *ip.IP6
	SubnetLen     uint
	IPv6SubnetLen uint
}
```

In theory, all could be called when using kube or subnet, even though most of them don't make any sense in kube. Probably we should add a warning or something when using kube that they don't make sense